lgc/interface/lgc/GpurtDialect.td

```diff
@@ -107,3 +109,183 @@ def GpurtGetTriangleCompressionModeOp : GpurtOp<"get.triangle.compression.mode",
+
+    - originY: Ray origin Y.
+
+    - originZ: Ray origin Z.
+
+    - tMin: T Min.
+
+    - dirX: World ray direction X.
+
+    - dirY: World ray direction Y.
+
+    - dirZ: World ray direction Z.
+  }];
+}
+
+def GpurtCallClosestHitShaderOp : GpurtOp<"call.closest.hit.shader", [WillReturn]> {
+  let arguments =  (ins V2I32:$shaderId, I32:$tableIndex);
+  let results = (outs I1:$result);
+
+  let summary = " Call closest hit shader";
+  let description = [{
+    Compiler closest hit shader inlining patch function.
+
+    - shaderId: Closest hit shader identifier.
+
+    - tableIndex: Hit group shader record table index.
+
+    Returns true if the call was inlined.
+  }];
+}
+
+def GpurtCallMissShaderOp : GpurtOp<"call.miss.shader", [WillReturn]> {
+  let arguments =  (ins V2I32:$shaderId, I32:$tableIndex);
+  let results = (outs I1:$result);
+
+  let summary = " Call miss shader";
+  let description = [{
+    Compiler miss shader inlining patch function.
+
+    - shaderId: Miss shader identifier.
+
+    - tableIndex: Miss shader record table index.
+
+    Returns true if the call was inlined.
+  }];
+}
+
+def GpurtCallTriangleAnyHitShaderOp : GpurtOp<"call.triangle.any.hit.shader", [WillReturn]> {
+  let arguments =  (ins V2I32:$shaderId, I32:$tableIndex, V2F32:$attr);
+  let results = (outs);
+
+  let summary = " Call any hit shader";
+  let description = [{
+    Compiler any hit shader inlining patch function.
+
+    - shaderId: AnyHit shader identifier.
+
+    - tableIndex: Hit group shader record index.
+
+    - attr: Triangle barycentrics.
+  }];
+}
+
+def GpurtCallIntersectionShaderOp : GpurtOp<"call.intersection.shader", [WillReturn]> {
+  let arguments =  (ins V2I32:$shaderId, V2I32:$anyHitShaderId, I32:$tableIndex);
+  let results = (outs);
+
+  let summary = " Call intersection shader";
+  let description = [{
+    Compiler intersection shader inlining patch function.
+
+    - shaderId: Programmable intersection shader identifier.
+
+    - anyHitShaderId: AnyHit shader identifier.
+
+    - tableIndex: Hit group shader record index.
+  }];
+}
+
+def GpurtSetTriangleIntersectionAttributesOp : GpurtOp<"set.triangle.intersection.attributes", [Memory<[(write)]>, WillReturn]> {
+  let arguments =  (ins V2F32:$barycentrics);
+  let results = (outs);
+
+  let summary = "Set triangle intersection attributes";
+  let description = [{
+    Update the current hit attributes for a triangle hit.
+
+    - barycentrics: Triangle barycentrics.
+  }];
+}
+
+def GpurtSetHitTriangleNodePointerOp : GpurtOp<"set.hit.triangle.node.pointer", [Memory<[(write)]>, WillReturn]> {
+  let arguments =  (ins I64:$bvhAddress, I32:$nodePointer);
+  let results = (outs);
+
+  let summary = "Set hit triangle node pointer";
+  let description = [{
+    Compiler notification of hit triangle node pointer.
+
+    - bvhAddress: The BVH address.
+
+    - nodePointer: Node pointer of hit triangle.
+  }];
+}
+
+def GpurtGetParentIdOp : GpurtOp<"get.parent.id", [WillReturn]> {
```

Comment:
Should be possible to make this one explicitly `Memory<[]>` because it doesn't touch mutable state, right?

Reply:
You're right, I missed that.